### PR TITLE
fix(Scripts/Dalaran): Update Toy Shop Toy Plane behaviour based on sniffs

### PR DESCRIPTION
**Pull request by sudlud**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

Thanks @heyitsbench for providing the sniffs!

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

.go c 105492
observe toy plane flying a circle below the ceiling, a cable connecting it to the hook at the ceiling

## Known Issues and TODO List:

- [ ] fix cyclic movementSpine core implementation to exclude original spawn point from loop
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/zone_dalaran.cpp [hunk 1]
```diff
@@ -795,6 +796,57 @@ class npc_dalaran_warrior : public CreatureScript
...
+    NPC_DND_DALARAN_TOY_STORE_PLANE_STRING_HOOK = 29807,
+
+    SPELL_TOY_PLANE_CABLE   = 55281,
+};
+
+struct npc_cosmetic_toy_plane : public ScriptedAI
+{
+    npc_cosmetic_toy_plane(Creature* creature) : ScriptedAI(creature)
+    {
+        Movement::MoveSplineInit init(me);
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/zone_dalaran.cpp:810]: Please move this to Reset(), avoid calling code inside the constructor (stuff may not be initialized at that point)

**Comment by kissingers** [comment]
I dont think it is a good way use scriptAI, just fix the SAI break link to next ID problem , might destroy by this PR, and so much problems will be fixed.

https://github.com/azerothcore/azerothcore-wotlk/pull/18026

**Review by Nyeriah — APPROVED** [review]

**Comment by elthehablo** [comment]
Why not do this in SmartAI using the waypoints table? See https://wowgaming.altervista.org/aowow/?npc=18670 in `smart_scripts` for an example. Or Little Noah in SW

**Comment by sudlud** [comment]
My first attempt was in SmartAI, but my conclusion was that SmartAI/waypoints does not support cyclic movementspine, so I went for the script.

TC / the cataclysm preservation project actually introduced a second waypoint table for cyclic movements, so "normal" waypoint movement and cyclic movement are just not the same and must be handled differently.

**Comment by sudlud** [comment]
So the short answer is, that it's currently not possible to implement the toy plane movement correctly in anything but a script imo.

**Comment by elthehablo** [comment]
Okay good answer. Thanks!

**Review by elthehablo — APPROVED** [review]

**Pull request merged**